go-controller/pkg/ovn/policy.go

```diff
@@ -95,17 +98,22 @@ func (oc *Controller) syncNetworkPolicies(networkPolicies []interface{}) {
 	}
 }
 
-func addAllowACLFromNode(logicalSwitch string, mgmtPortIP net.IP) error {
+func addAllowACLFromNode(logicalSwitch string, mgmtPortIP net.IP, ovnNBClient goovn.Client) error {
 	ipFamily := "ip4"
 	if utilnet.IsIPv6(mgmtPortIP) {
 		ipFamily = "ip6"
 	}
 	match := fmt.Sprintf("%s.src==%s", ipFamily, mgmtPortIP.String())
-	_, stderr, err := util.RunOVNNbctl("--may-exist", "acl-add", logicalSwitch,
-		"to-lport", defaultAllowPriority, match, "allow-related")
-	if err != nil {
-		return fmt.Errorf("failed to create the node acl for "+
-			"logical_switch=%s, stderr: %q (%v)", logicalSwitch, stderr, err)
+
+	priority, _ := strconv.Atoi(defaultAllowPriority)
+	aclcmd, err := ovnNBClient.ACLAdd(logicalSwitch, "to-lport", match, "allow-related", priority, nil, false, "", "")
+	if err == nil {
+		// NOTE: goovn.ErrorExist is returned if the ACL already exists, skip returning an error
+		if err = ovnNBClient.Execute(aclcmd); err != nil && err != goovn.ErrorExist {
+			return fmt.Errorf("failed to create the node acl for logical_switch: %s, %v", logicalSwitch, err)
```

Comment:
Why do we want to ignore the ErrorExist error?  The lookup is only being done on the direction, match, and priority fields.  If the user was trying to create an ACL that contained different values for other fields (e.g., action), they wouldn't know that didn't happen.

Reply:
I am maintaining the equivalency of the original code, refer https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/policy.go#L104-L105 ,( note the `--may-exist` option)

Reply:
We usually allow for "may-exist" type of behavior because if ovnkube dies, when it restarts it will get all of the events from kubernetes and try to re-run all of this which may still already be present in the NBDB. So we try to keep things idempotent. We've had some discussion in the past about handling this better by maintaining and ovn-k8s state within CRDs or some other DB mechanism, but its kind of further down the road.

Reply:
@anfredette ah I see what you mean. Sorry I missed it. I presumed the may-exist would also check action for ACL equivalency. Should this be something fixed in go-ovn or OVN? For now should we find the ACL from the go-ovn cache and compare the action as a workaround to be 100% sure it is the same as what we are intending?

Reply:
If we want to be 100% sure, then I think that's what we'd need to do.  However, if we know we always create them a certain way, then maybe we could skip the comparison.  That's not the way the APIs work currently, and I don't think Han would be ok with changing the semantics.  Perhaps an option is to add a new behavior, like --may-exist, but we could call it something else like --force, which would change the ACL to have all the properties in the create.  The new ACL APIs I added haven't been there that long.  Han may be okay with updating them.

Reply:
I think the way the new API could work is:
- if the ACL isn't found, just create it as normal.
- if the ACL is found, then compare the rest of the fields and update any that are different.

Reply:
Seems reasonable to me. For now we can push this PR through as the chance of this happening with the changes in this patch is pretty much none. Let's see how the discussion in go-ovn goes.

Reply:
I'd have to double-check, but I think the current code using RunOVNNbctl would have the same issue.